wagtail/wagtailadmin/tests/test_pages_views.py

```diff
@@ -1474,6 +1475,29 @@ def test_search(self):
         self.assertTemplateUsed(response, 'wagtailadmin/pages/search.html')
         self.assertEqual(response.context['query_string'], "Hello")
 
+    def test_search_searchable_fields(self):
+        # Find root page
+        root_page = Page.objects.get(id=2)
+
+        # Create a page
+        root_page.add_child(instance=SimplePage(
+            title="Hi there!",
+            slug='hello-world',
+            live=True,
+            has_unpublished_changes=False,
+        ))
+
+        # Confirm the slug is not being searched
+        response = self.get({'q': "hello"})
+        self.assertNotContains(response, "There is one matching page")
+
+        # Add slug to the search_fields
+        Page.search_fields = Page.search_fields + (SearchField('slug', partial_match=True),)
```

Comment:
I think this change will persist after the test has completed, which could cause unexpected behaviour in later tests. Saving the initial value of `search_fields` and restoring it at the end of the test should be enough to solve this, though.